api-server/scanner-daemon/src/main.rs

```diff
@@ -88,6 +88,19 @@ pub async fn run<S: ApiServerStorage>(
                 .expect("Can't scan genesis");
             local_block
         } else {
+            let storage_version = db_tx
+                .get_storage_version()
+                .await
+                .unwrap_or_else(|e| panic!("Storage version read failed {}", e))
+                .expect("cannot be empty");
+
+            if storage_version != CURRENT_STORAGE_VERSION {
+                db_tx
+                    .reinitialize_storage(chain_config)
+                    .await
+                    .unwrap_or_else(|e| panic!("Storage re-initialization failed {}", e));
```

Comment:
Can we do better than panic here? Preferably, panic would be used for internal invariant violations. But here it seems it can happen due to external factors (e.g. disk full).

Reply:
I see that we don't care panicing or expecting everywhere across api-server

Reply:
Ok, I guess it may be fine here because storage migration is a sort of one-off event.